In [6]:
import pandas as pd
from donnees_IDFM import donnees_stations_toutes, donnees_validation_23

In [7]:
data=donnees_validation_23()

In [8]:
stations=donnees_stations_toutes()

In [9]:
data.head()

JOUR  CODE_STIF_TRNS CODE_STIF_RES  ...    lda CATEGORIE_TITRE  NB_VALD
0  2023-06-26             800           853  ...  67000          NAVIGO      938
1  2023-06-26             800           853  ...  67000      NON DEFINI       71
2  2023-06-26             800           853  ...  67227     NAVIGO JOUR        5
3  2023-06-26             800           853  ...  67126               ?        5
4  2023-06-26             800           853  ...  67292               ?      147

[5 rows x 8 columns]

In [10]:
stations.head()

Geo Point  ...                        geometry
0   49.04516945934257, 2.322917846227108  ...  POINT (650500.365 6883022.317)
1   48.53289010230884, 2.290082513840396  ...  POINT (647587.834 6826086.019)
2    48.7510232741848, 2.215860857351649  ...  POINT (642349.149 6850387.489)
3  48.529746917640566, 2.572073662735742  ...  POINT (668404.662 6825586.613)
4  48.725535303926115, 2.419565024387933  ...  POINT (657304.576 6847424.318)

[5 rows x 28 columns]

In [18]:
result = pd.merge(data, stations, left_on='lda', right_on='id_ref_ZdC', how='outer')

In [19]:
result[result['lda'] == 63025]

JOUR  CODE_STIF_TRNS  ...      nom_iv                        geometry
328333  2023-06-26           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
328334  2023-06-26           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
328335  2023-05-12           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
328336  2023-05-12           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
328337  2023-05-12           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
...            ...             ...  ...         ...                             ...
329748  2023-04-29           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
329749  2023-04-29           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
329750  2023-06-28           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
329751  2023-06-30           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)
329752  2023-06-26           810.0  ...  Le Guichet  POINT (640597.105 6845332.816)

[1420 rows x 36 columns]

In [21]:
result.head()

JOUR  ...                        geometry
0  2023-06-26  ...  POINT (650520.696 6885371.019)
1  2023-06-26  ...  POINT (650520.696 6885371.019)
2  2023-05-03  ...  POINT (650520.696 6885371.019)
3  2023-05-03  ...  POINT (650520.696 6885371.019)
4  2023-05-03  ...  POINT (650520.696 6885371.019)

[5 rows x 36 columns]

In [22]:
data.shape

(1096209, 8)

In [24]:
result.shape

(1543107, 36)

In [25]:
result_RER=result[result['mode_'] == 'RER']

In [26]:
result_RER.shape

(375851, 36)

In [27]:
result_RER.head()

JOUR  CODE_STIF_TRNS  ...      nom_iv                        geometry
5247  2023-06-26           800.0  ...  Pierrelaye  POINT (638169.176 6880245.881)
5249  2023-06-26           800.0  ...  Pierrelaye  POINT (638169.176 6880245.881)
5251  2023-05-12           800.0  ...  Pierrelaye  POINT (638169.176 6880245.881)
5253  2023-05-12           800.0  ...  Pierrelaye  POINT (638169.176 6880245.881)
5255  2023-05-03           800.0  ...  Pierrelaye  POINT (638169.176 6880245.881)

[5 rows x 36 columns]

In [31]:
result_RER['LIBELLE_ARRET'].nunique()

248

In [32]:
result_RER['id_ref_ZdC'].nunique()

244

In [33]:
result_RER[result_RER['LIBELLE_ARRET'].isna()]

JOUR  ...                        geometry
1542818  NaN  ...  POINT (649716.273 6869205.791)
1542822  NaN  ...  POINT (652093.225 6862619.067)
1542823  NaN  ...  POINT (652093.225 6862619.067)
1542824  NaN  ...  POINT (652093.225 6862619.067)
1542865  NaN  ...  POINT (660596.964 6898729.346)
1542877  NaN  ...  POINT (637836.169 6815513.623)
1542878  NaN  ...  POINT (654572.589 6843467.919)
1542879  NaN  ...  POINT (654645.086 6843445.410)
1542880  NaN  ...  POINT (655584.693 6799419.830)
1542929  NaN  ...  POINT (664015.187 6891935.660)
1542936  NaN  ...  POINT (662795.040 6893286.621)
1542941  NaN  ...  POINT (652885.802 6864829.589)
1542942  NaN  ...  POINT (652885.802 6864829.589)
1542943  NaN  ...  POINT (652970.185 6864719.450)
1542994  NaN  ...  POINT (677235.313 6852342.427)
1542999  NaN  ...  POINT (661337.234 6907273.654)
1543037  NaN  ...  POINT (666964.707 6860182.508)
1543101  NaN  ...  POINT (650849.519 6863778.763)

[18 rows x 36 columns]

In [35]:
result_RER[result_RER['nom_long'].isna()]

Empty DataFrame
Columns: [JOUR, CODE_STIF_TRNS, CODE_STIF_RES, CODE_STIF_ARRET, LIBELLE_ARRET, lda, CATEGORIE_TITRE, NB_VALD, Geo Point, Geo Shape, gares_id, nom_long, nom_so_gar, nom_su_gar, id_ref_ZdC, nom_ZdC, id_ref_ZdA, nom_ZdA, idrefliga, idrefligc, res_com, indice_lig, mode_, tertrain, terrer, termetro, tertram, terval, exploitant, idf, principal, x, y, picto ligne, nom_iv, geometry]
Index: []

In [38]:
result_RER[result_RER['res_com'].isna()]

Empty DataFrame
Columns: [JOUR, CODE_STIF_TRNS, CODE_STIF_RES, CODE_STIF_ARRET, LIBELLE_ARRET, lda, CATEGORIE_TITRE, NB_VALD, Geo Point, Geo Shape, gares_id, nom_long, nom_so_gar, nom_su_gar, id_ref_ZdC, nom_ZdC, id_ref_ZdA, nom_ZdA, idrefliga, idrefligc, res_com, indice_lig, mode_, tertrain, terrer, termetro, tertram, terval, exploitant, idf, principal, x, y, picto ligne, nom_iv, geometry]
Index: []

In [40]:
result[result['lda'] == 73794]

JOUR  CODE_STIF_TRNS CODE_STIF_RES  ... picto ligne nom_iv  geometry
640248  2023-06-27           100.0           110  ...         NaN    NaN      None
640249  2023-06-27           100.0           110  ...         NaN    NaN      None
640250  2023-06-27           100.0           110  ...         NaN    NaN      None
640251  2023-05-12           100.0           110  ...         NaN    NaN      None
640252  2023-05-12           810.0           801  ...         NaN    NaN      None
...            ...             ...           ...  ...         ...    ...       ...
643501  2023-06-30           100.0           110  ...         NaN    NaN      None
643502  2023-06-30           810.0           801  ...         NaN    NaN      None
643503  2023-06-30           810.0           801  ...         NaN    NaN      None
643504  2023-06-26           810.0           801  ...         NaN    NaN      None
643505  2023-06-27           100.0           110  ...         NaN    NaN      None

[3258 rows x 36 columns]

CCL : pas de problème pour les élements de la jointure avec RER
mais il faut identifier les pb si on enlève RER car pas exemple chatelet il manque un numéro

In [51]:
result[result['mode_'].isna()]

JOUR  CODE_STIF_TRNS CODE_STIF_RES  ... picto ligne nom_iv  geometry
150461   2023-06-26           800.0           854  ...         NaN    NaN      None
150462   2023-05-12           800.0           854  ...         NaN    NaN      None
150463   2023-05-03           800.0           854  ...         NaN    NaN      None
150464   2023-05-04           800.0           854  ...         NaN    NaN      None
150465   2023-05-07           800.0           854  ...         NaN    NaN      None
...             ...             ...           ...  ...         ...    ...       ...
1541725  2023-06-28           800.0           851  ...         NaN    NaN      None
1541726  2023-06-04           800.0           851  ...         NaN    NaN      None
1541727  2023-06-30           800.0           851  ...         NaN    NaN      None
1541728  2023-06-30           800.0           851  ...         NaN    NaN      None
1541729  2023-06-30           800.0           851  ...         NaN    NaN      None

[37939 rows x 36 columns]

In [52]:
result[result['mode_'].isna()].nunique()

JOUR                181
CODE_STIF_TRNS        3
CODE_STIF_RES         9
CODE_STIF_ARRET      26
LIBELLE_ARRET        26
lda                  24
CATEGORIE_TITRE       9
NB_VALD            6460
Geo Point             0
Geo Shape             0
gares_id              0
nom_long              0
nom_so_gar            0
nom_su_gar            0
id_ref_ZdC            0
nom_ZdC               0
id_ref_ZdA            0
nom_ZdA               0
idrefliga             0
idrefligc             0
res_com               0
indice_lig            0
mode_                 0
tertrain              0
terrer                0
termetro              0
tertram               0
terval                0
exploitant            0
idf                   0
principal             0
x                     0
y                     0
picto ligne           0
nom_iv                0
geometry              0
dtype: int64

In [58]:
result[result['mode_'].isna()]['LIBELLE_ARRET'].drop_duplicates().shape

(26,)

In [57]:
result[result['mode_'].isna()]['LIBELLE_ARRET'].drop_duplicates()

150461     TRIEL SUR SEINE
262178      NOISY-LE-GRAND
614519         LA CHAPELLE
614521             MAGENTA
627380           LA MUETTE
632197               AVRON
636992         LE PELETIER
640248          LES HALLES
640252            CHATELET
703089            BASTILLE
747300      CHATEAU-LANDON
758445      ETIENNE MARCEL
777853        MONTPARNASSE
837570      PORTE DAUPHINE
866316      PORTE D.CLIGN.
953502          BIR-HAKEIM
1063134            Inconnu
1126763             JUVISY
1133654         GRESILLONS
1257188    OZOIR LA FERRIE
1403670              AUBER
1448484            ETAMPES
1483777          SEVRES RG
1486169            TRAPPES
1491604           EPLUCHES
1541262              BERCY
Name: LIBELLE_ARRET, dtype: object

Voici donc les 26 stations sur l'ensemble du réseau avec un problème de jointure. _A priori_ une petite dizaine sur le RER (à confirmer).

## Identification des stations avec plusieurs lda
On cherche les stations pour lesquelles il y a plusieurs lda pour une même station ou un même lda pour deux stations

In [59]:
unique_lda_check = data.groupby('LIBELLE_ARRET')['lda'].nunique()

# Afficher les LIBELLE_ARRET avec plus d'un lda
non_unique_lda = unique_lda_check[unique_lda_check > 1]
print(non_unique_lda)

LIBELLE_ARRET
BERCY           2
CHATELET        2
MONTPARNASSE    2
Name: lda, dtype: int64


In [66]:
unique_lda_check = data.groupby('lda')['LIBELLE_ARRET'].nunique()

# Afficher les LIBELLE_ARRET avec plus d'un lda
non_unique_lda = unique_lda_check[unique_lda_check > 1]
print(non_unique_lda)

lda
63244    2
63320    2
64382    2
64589    2
64622    2
65227    2
70956    2
71135    2
71150    2
71370    2
71453    2
71517    2
71545    2
71572    2
72126    2
72383    2
72641    2
72787    2
73620    2
73661    2
73731    2
73794    2
73844    2
74000    2
Name: LIBELLE_ARRET, dtype: int64


71150 Mirabeau et Javel !!!

73661 Brochant et pont cardinet

73794 Chatelet et les Halles !!!

73844 La motte et Champ de Mars !!

74000 La chappelle et Magenta

In [98]:
lda=non_unique_lda.to_frame()

In [99]:
lda = lda.index.tolist()

In [100]:
for nb in lda:
    libelle_arret_unique = data[data['lda'] == nb]['LIBELLE_ARRET'].drop_duplicates()
    print(f"LIBELLE_ARRET pour lda {nb}:")
    print(libelle_arret_unique, "\n")


LIBELLE_ARRET pour lda 63244:
233       MASSY-PALAIS.
2456    MASSY PALAISEAU
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 63320:
235      MASSY-VERRIER.
1211    MASSY VERRIERES
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 64382:
93                               ST NOM LA BRETE
1141    Saint-Nom-la-Bretèche Forêt de Marly T13
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 64589:
182                    ST-GERMAIN
1139    Saint-Germain-en-Laye T13
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 64622:
111          ST GERMAIN GCO
1143    Lisière Pereire T13
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 65227:
1126     EPINAY VIL T11
1497    EPINAY VILLETAN
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 70956:
74     NANTERRE UNIVER
156     NANTERRE-UNIV.
Name: LIBELLE_ARRET, dtype: object 

LIBELLE_ARRET pour lda 71135:
409     GARE D'AUSTER.
1191        AUSTERLITZ
Name: LIBELLE_ARRET, dtype: object 

LIBELL

Il faut maintenant voir uniquement pour les RER